src/JSObjectProxy.cc

```diff
@@ -0,0 +1,276 @@
+/**
+ * @file JSObjectProxy.cc
+ * @author Caleb Aikens (caleb@distributive.network) & Tom Tang (xmader@distributive.network)
+ * @brief JSObjectProxy is a custom C-implemented python type that derives from dict. It acts as a proxy for JSObjects from Spidermonkey, and behaves like a dict would.
+ * @version 0.1
+ * @date 2023-06-26
+ *
+ * Copyright (c) 2023 Distributive Corp.
+ *
+ */
+
+#include "include/JSObjectProxy.hh"
+
+#include "include/modules/pythonmonkey/pythonmonkey.hh"
+#include "include/jsTypeFactory.hh"
+#include "include/pyTypeFactory.hh"
+
+#include <jsapi.h>
+#include <jsfriendapi.h>
+#include <js/Equality.h>
+
+#include <Python.h>
+
+JSContext *GLOBAL_CX; /**< pointer to PythonMonkey's JSContext */
+
+void JSObjectProxyMethodDefinitions::JSObjectProxy_dealloc(JSObjectProxy *self)
+{
+  // TODO (Caleb Aikens): intentional override of PyDict_Type's tp_dealloc. Probably results in leaking dict memory
+  return;
+}
+
+PyObject *JSObjectProxyMethodDefinitions::JSObjectProxy_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+  JSObjectProxy *self;
+  self = (JSObjectProxy *)type->tp_alloc(type, 0);
+  if (!self)
+  {
+    Py_DECREF(self);
+    return NULL;
+  }
+  self->jsObject.set(JS_NewObject(GLOBAL_CX, NULL));
+  return (PyObject *)self;
+}
+
+int JSObjectProxyMethodDefinitions::JSObjectProxy_init(JSObjectProxy *self, PyObject *args, PyObject *kwds)
+{
+
+  PyObject *dict = NULL;
+  if (PyTuple_Size(args) == 0 || Py_IsNone(PyTuple_GetItem(args, 0))) {
```

Comment:
```suggestion
  if (PyTuple_Size(args) == 0 || PyTuple_GetItem(args, 0) == Py_None) {
```
`Py_IsNone` does not exist on Python versions < 3.10
https://docs.python.org/3/c-api/structures.html#c.Py_IsNone